File: notely_api/app/models/note.py
from datetime import datetime, timezone
from typing import Optional

from beanie import Document, Indexed, PydanticObjectId
from pydantic import Field


class Note(Document):
    user_id: Indexed(PydanticObjectId)
    title: str
    content: str
    tags: list[str] = Field(default_factory=list)
    source_url: Optional[str] = None
    summary_text: Optional[str] = None
    summary_model: Optional[str] = None
    summary_generated_at: Optional[datetime] = None
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

    class Settings:
        name = "notes"

File: notely_app/lib/models/note.dart
class Note {
  final String id;
  final String title;
  final String content;
  final List<String> tags;
  final String? sourceUrl;
  final String? summaryText;
  final DateTime createdAt;
  final DateTime updatedAt;

  Note({
    required this.id,
    required this.title,
    required this.content,
    required this.tags,
    this.sourceUrl,
    this.summaryText,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Note.fromJson(Map<String, dynamic> json) {
    return Note(
      id: json["id"],
      title: json["title"],
      content: json["content"],
      tags: List<String>.from(json["tags"]),
      sourceUrl: json["source_url"],
      summaryText: json["summary_text"],
      createdAt: DateTime.parse(json["created_at"]),
      updatedAt: DateTime.parse(json["updated_at"]),
    );
  }
}
